Assets/Mirror/Runtime/NetworkIdentity.cs

```diff
@@ -1067,53 +1067,54 @@ public void RebuildObservers(bool initialize)
         }
 
         /// <summary>
-        /// Removes ownership for an object for a client by its connection.
-        /// <para>This applies to objects that had authority set by AssignClientAuthority, or NetworkServer.SpawnWithClientAuthority. Authority cannot be removed for player objects.</para>
+        /// Obsolete: Use <see cref="RemoveClientAuthority()"/> instead
         /// </summary>
         /// <param name="conn">The connection of the client to remove authority for.</param>
         /// <returns>True if authority is removed.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never), Obsolete("NetworkConnection parameter is no longer needed and nothing is returned")]
         public bool RemoveClientAuthority(NetworkConnection conn)
+        {
+            RemoveClientAuthority();
+            return true;
+        }
+
+        /// <summary>
+        /// Removes ownership for an object.
+        /// <para>This applies to objects that had authority set by AssignClientAuthority, or NetworkServer.SpawnWithClientAuthority.</para>
+        /// <para>Authority cannot be removed for player objects.</para>
+        /// </summary>
+        public void RemoveClientAuthority()
         {
             if (!isServer)
             {
                 Debug.LogError("RemoveClientAuthority can only be call on the server for spawned objects.");
-                return false;
+                return;
             }
 
             if (connectionToClient != null)
             {
                 Debug.LogError("RemoveClientAuthority cannot remove authority for a player object");
-                return false;
+                return;
             }
 
-            if (clientAuthorityOwner == null)
+            if (clientAuthorityOwner != null)
             {
-                Debug.LogError("RemoveClientAuthority for " + gameObject + " has no clientAuthority owner.");
-                return false;
-            }
+                // send msg to that client
+                ClientAuthorityMessage msg = new ClientAuthorityMessage
+                {
+                    netId = netId,
+                    authority = false
+                };
 
-            if (clientAuthorityOwner != conn)
-            {
-                Debug.LogError("RemoveClientAuthority for " + gameObject + " has different owner.");
-                return false;
+                clientAuthorityOwner.Send(msg);
+                clientAuthorityCallback?.Invoke(clientAuthorityOwner, this, false);
```

Comment:
this is now being called before ForceAuthority(true).   Does this break something?

Reply:
Seems to work fine, and needs to be before clientAuthorityOwner is set null.  I don't think it matters but I think the message should get out on the wire first.

Reply:
well,  the behavior changes a bit, now the event handler here will still see the clientAuthorityOwner,  before that,  it saw null.